HARK/ConsumptionSaving/tests/test_ConsPortfolioModel.py

```diff
@@ -39,37 +39,47 @@ def test_solution(self):
     def test_sim_one_period(self):
         self.pcct.T_sim = 30
         self.pcct.AgentCount = 10
-        self.pcct.track_vars += ["aNrm"]
+        self.pcct.track_vars += ["aNrm", "mNrm", "bNrm", "TranShk", "cNrm"]
         self.pcct.initialize_sim()
 
         self.assertFalse(np.any(self.pcct.shocks["Adjust"]))
 
-        self.assertAlmostEqual(self.pcct.state_now["pLvl"][0], 1.0)
+        # simulation test -- seed/generator specific
+        # self.assertAlmostEqual(self.pcct.state_now["pLvl"][0], 1.0)
 
-        self.assertAlmostEqual(self.pcct.state_now["aNrm"][0], 7.257027956)
+        # simulation test -- seed/generator specific
+        # self.assertAlmostEqual(self.pcct.state_now["aNrm"][0], 7.257027956)
 
-        self.assertAlmostEqual(self.pcct.Rfree[0], 1.03)
+        # simulation test -- seed/generator specific
+        #self.assertAlmostEqual(self.pcct.Rfree[0], 1.03)
 
-        self.assertAlmostEqual(self.pcct.state_now["PlvlAgg"], 1.0)
+        # simulation test -- seed/generator specific
+        #self.assertAlmostEqual(self.pcct.state_now["PlvlAgg"], 1.0)
 
         self.pcct.sim_one_period()
 
-        self.assertAlmostEqual(self.pcct.state_now["pLvl"][0], 0.858934461)
-
-        self.assertAlmostEqual(self.pcct.state_now["aNrm"][0], 8.02359093)
-
-        self.assertAlmostEqual(self.pcct.shocks["PermShk"][0], 0.85893446)
-
-        self.assertAlmostEqual(self.pcct.shocks["TranShk"][0], 1.0)
+        self.assertAlmostEqual(
```

Comment:
These are what you have called 'functional' tests.

I want to ask what kind of error you see these catching.

The only way I can see this failing is someone messing up the transition equations of the simulation method, or introducing a bug in the time indexing of `hark.core` so that the states and shocks get badly out of sync. Both are valuable, I just want to know if you see other possible cases.

Reply:
These are the generalization of the specific value test that makes the fewest assumptions.
I think you have pointed out two ways the test could fail.

I've been thinking about it but can't come up with others. Why do you ask?